pkg/compiler/codegen.go

```diff
@@ -1694,6 +1694,9 @@ func transformArgs(fun ast.Expr, args []ast.Expr) []ast.Expr {
 		if f.Sel.Name == "FromAddress" {
 			return args[1:]
 		}
+		if strings.HasPrefix(f.Sel.Name, "Syscall") {
```

Comment:
Won't this affect user-defined `Syscall*` functions?

Reply:
Fixed.